modules/avd/avd.bicep

```diff
@@ -119,3 +132,156 @@ resource workspace 'Microsoft.DesktopVirtualization/workspaces@2019-12-10-previe
 // }
 
 output appGroupName string = appGroupName
+
+
+
+// // Azure Virtual Desktop and Session Hosts region
+
+resource nic 'Microsoft.Network/networkInterfaces@2020-06-01' = [for i in range(0, countAVDInstances): {
+  name: 'vm-redcap-${i}-nic'
+  location: location
+  properties: {
+    ipConfigurations: [
+      {
+        name: 'ipconfig'
+        properties: {
+          privateIPAllocationMethod: 'Dynamic'
+          subnet: {
+            id: ComputeSubnetId
+          }
+        }
+      }
+    ]
+  }
+}]
+
+resource vm 'Microsoft.Compute/virtualMachines@2023-03-01' = [for i in range(0, countAVDInstances): {
+  name: 'vm-redcap-${i}'
+  location: location
+  identity: AADJoin ? {
+    type: 'SystemAssigned'
+  } : null
+  properties: {
+    licenseType: 'Windows_Client'
+    hardwareProfile: {
+      vmSize: vmSku
+    }
+    osProfile: {
+      computerName: 'vm-redcap-${i}'
+      adminUsername: avdVMAdmin
+      adminPassword: avdVMPassword
+      windowsConfiguration: {
+        enableAutomaticUpdates: false
+        patchSettings: {
+          patchMode: 'Manual'
+        }
+      }
+    }
+    storageProfile: {
+      osDisk: {
+        name: 'vm-OS-${i}'
+        caching: vmDiskCachingType
+        managedDisk: {
+          storageAccountType: vmDiskType
+        }
+        osType: 'Windows'
+        createOption: 'FromImage'
+      }
+      // TODO Turn into params
+      imageReference: {
+        publisher: 'microsoftwindowsdesktop'
+        offer: 'office-365'
+        sku: '20h2-evd-o365pp'
+        version: 'latest'
+      }
+      dataDisks: []
+    }
+    networkProfile: {
+      networkInterfaces: [
+        {
+          id: nic[i].id
+        }
+      ]
+    }
+  }
+  dependsOn: [
+    nic[i]
+  ]
+}]
+
+resource domainJoinExtension 'Microsoft.Compute/virtualMachines/extensions@2018-10-01' = [for i in range(0, countAVDInstances): {
+  name: 'AADJoin'
+  parent: vm[i]
+  location: location
+  properties: AADJoin ? {
```

Comment:
Move this conditional to the resource level. Otherwise we're creating resources that don't do anything.

`resource aadJoinExtension 'Microsoft.Compute/virtualMachines/extensions@2018-10-01' = [for i in range(0, countAVDInstances): if (AADJoin) {`